File: src/main/com/lec03/Lec03Main.java
package main.com.lec03;

import main.com.etc.java.JavaPerson;

public class Lec03Main {
    public static void main(String[] args) {
        JavaPerson javaPerson = new JavaPerson("류욱상", 100);
        String log = String.format(
                "사람의 이름은 %s이고 나이는 %s세 입니다.",
                javaPerson.getName(),
                javaPerson.getAge()
        );

        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("사람의 이름은");
        stringBuilder.append(javaPerson.getName());
        stringBuilder.append("이고 나이는");
        stringBuilder.append(javaPerson.getAge());
        stringBuilder.append("세 입니다.");

        String str = "ABCDE";
        char ch = str.charAt(1);
    }

    public static void printAgeIfPerson(Object obj) {
        if (obj instanceof JavaPerson) {
            JavaPerson javaPerson = (JavaPerson) obj;
            System.out.println(javaPerson.getAge());
        }
    }
}

File: src/main/com/lec03/Lec03Main.kt
package main.com.lec03

import main.com.etc.JavaPerson

fun main() {
    val number1 = 3
    val number2: Long = number1.toLong()

    val number3: Int? = 3
    val number4: Long = number3?.toLong() ?: 0

    printAgeIfPerson2(JavaPerson("류욱상"))
    printAgeIfPerson2(JavaPerson("류욱상", 100))

    val javaPerson = JavaPerson("류욱상", 100)
    val log = "사람의 이름은 ${javaPerson.name}이고 나이는 ${javaPerson.age}세 입니다"

    val name = javaPerson.name
    val age = javaPerson.age
    val log2 = "사람의 이름은 $name 이고 나이는 $age 입니다."

    println(log)
    println(log2)

    val str = """
        ABC
        DEF
        ${name}
    """.trimIndent()
    println(str)

    val str2 = "ABCDE"
    val ch = str2[1]
    println(ch)
}

fun printAgeIfPerson(obj: Any) {
    if (obj is JavaPerson) {
        val javaPerson = obj as JavaPerson
        println(javaPerson.age)
    }
}

fun printAgeIfPerson2(obj: Any?) {
    val javaPerson = obj as? JavaPerson
    println(javaPerson?.age)
}

fun fail(message: String): Nothing {
    throw IllegalArgumentException(message)
}